plugins/modules/cloudtrail_info.py

```diff
@@ -0,0 +1,235 @@
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ansible_dict_to_boto3_filter_list
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+
+
+def get_trails(connection, module):
+    all_trails = []
+    try:
+        result = connection.get_paginator('list_trails')
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Failed to get the trails.")
+    for trail in result.paginate():
+        all_trails.extend(list_cloud_trails(trail))
+    return all_trails
+
+
+def list_cloud_trails(trail_dict):
+    return [x["TrailARN"] for x in trail_dict["Trails"]]
+
+
+def get_trail_detail(connection, module, trail_name_list=None, include_shadow_trails=True):
+    if not trail_name_list:
+        trail_name_list = get_trails(connection, module)
+    try:
+        result = connection.describe_trails(trailNameList=trail_name_list, includeShadowTrails=True, aws_retry=True)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Failed to get the trails.")
+    # Turn the boto3 result in to ansible_friendly_snaked_names
+    snaked_cloud_trail = []
+    for cloud_trail in result['trailList']:
+        try:
+            status_dict = connection.get_trail_status(Name=cloud_trail["TrailARN"], aws_retry=True)
+            cloud_trail.update(status_dict)
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e, msg="Failed to get the trail status")
+        try:
+            tag_list = connection.list_tags(ResourceIdList=[cloud_trail["TrailARN"]])
+            for tag_dict in tag_list["ResourceTagList"]:
+                cloud_trail.update(tag_dict)
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            module.warn("Failed to get the trail tags - {0}".format(e))
+        snaked_cloud_trail.append(camel_dict_to_snake_dict(cloud_trail))
+
+    # Turn the boto3 result in to ansible friendly tag dictionary
+    for tr in snaked_cloud_trail:
+        if 'tags_list' in tr:
+            tr['tags'] = boto3_tag_list_to_ansible_dict(tr['tags_list'], 'key', 'value')
+            del (tr['tags_list'])
+        if 'response_metadata' in tr:
+            del (tr['response_metadata'])
+
+    result['trail_list'] = snaked_cloud_trail
+    return result
+
+
+def main():
+    argument_spec = dict(
+        trail_names=dict(type='list', elements='str', default=[]),
+        include_shadow_trails=dict(type='bool', default=True),
+    )
```

Comment:
The module.params are never used, `get_trails()` retrieves all trails and `includeShadowTrails` is hard-coded as True in a few places.
Could you please assign `module.params.get('trail_names')` and `module.params.get('include_shadow_trails')` to variables either in `main()` or in `get_trail_detail()` and pass those into the different functions and API calls as needed?